Jenkinsfile

```diff
@@ -89,13 +89,31 @@ pipeline {
                 }
             }
         }
+        stage("Running Java tests") {
+            steps {
+                script {
+                    sh "docker exec -u saluser \${container_name} sh -c \"" +
+                        "source ~/.setup.sh && " +
+                        "export LSST_DDS_QOS=file:///home/saluser/repos/ts_ddsconfig/qos/QoS.xml && " +
+                        "salgenerator Test validate && " +
+                        "salgenerator Test sal java && " +
+                        "salgenerator Test maven && " +
+                        "salgenerator Script validate && " +
+                        "salgenerator Script sal java && " +
+                        "salgenerator Script maven && " +
+                        "cd /home/saluser/repos/ts_sal/java_tests && " +
+                        "mvn test\""
```

Comment:
Before running the `mvn test` you need to setup `LSST_DDS_PARTITION_PREFIX`, because it is not set by default in the develop-env. I would suggest doing something like:

```
export LSST_DDS_PARTITION_PREFIX=$(python -c "import os; from lsst.ts.salobj import set_random_lsst_dds_partition_prefix; set_random_lsst_dds_partition_prefix(); print(os.environ[\"LSST_DDS_PARTITION_PREFIX\"])")
```

Or some other dynamic `LSST_DDS_PARTITION_PREFIX`, so tests won't interfere with each other.

Reply:
I the Java tests aren't setting this env var to *something* that sounds like a misfeature in the tests. Note that they don't change it per test, which is also probably a misfeature.

Reply:
@wvreeven are you setting LSST_DDS_PARTITION_PREFIX? Please remind me why you don't set it to a new value for each test. It is only read when the SAL object is created, so changing it per test is reasonable (no matter how persistent the JVM's state is, unless it literally prevents changing it on the fly).

Reply:
@pkubanek I don't see any sign of the C++ tests setting LSST_DDS_PARTITION_PREFIX. Do they? It would be best to set it to a new random value before constructing the SAL object(s) for each test case. That prevents tests from interfering with each other.

Reply:
Java doesn't allow for modifying the values of system environment variables. The only way to do it is to mock the System library, which also is used for loading the native libraries so it is way too complicated to do so. In a few cases mocking the System library in the SAL Java unit tests is possible and there it is done. In all other cases, which means most of them, it is not done.

Reply:
For now I'l do the quick and easy change to Jenkins. I hope we can improve the C++ and Java tests in another ticket.

Reply:
@wvreeven @dmills-lsst I'll file a ticket asking that the Java SAL API support specifying an (optional) partition_prefix when constructing the SAL object. If omitted then the env variable can be used. This should work around the problem that Java cannot modify the environment.

Reply:
As to doing it in bash...that's fine in some cases, and is better than what we are doing now in the Jenkinsfile.  But I don't see how it can isolate the individual Java test cases.